File: Leetcode Challenge/March/Python/Rotate List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        if(head == None) :
            return head;
        
        
        len=1
        newH=tail=head;
        
        while tail.next != None:  
            tail = tail.next;
            len+=1;
        
        tail.next = head;

        k = k % len;
        
        for i in range(0, len-k):
            tail = tail.next
        newH = tail.next; 
        tail.next = None;
        return newH;

File: Leetcode Challenge/March/Java/Rotate List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if(head == null) {
            return head;
        }
        
        int len=1; 
        ListNode newH, tail;
        newH=tail=head;
        
        while(tail.next != null)  
        {
            tail = tail.next;
            len++;
        }
        tail.next = head;

        k = k % len;
        
        for(int i=0; i<len-k; i++) tail = tail.next; 
        newH = tail.next; 
        tail.next = null;
        return newH;
    }
}
